packages/fury-adapter-oas3-parser/lib/parser/oas/parseOauthFlowsObject.js

```diff
@@ -0,0 +1,88 @@
+const R = require('ramda');
+const {
+  isAnnotation, isExtension, hasKey,
+} = require('../../predicates');
+const {
+  createWarning,
+  createInvalidMemberWarning,
+} = require('../annotations');
+const pipeParseResult = require('../../pipeParseResult');
+const parseObject = require('../parseObject');
+const parseOauthFlowObject = require('./parseOauthFlowObject');
+
+const name = 'Oauth Flows Object';
+
+const validFlow = R.anyPass(R.map(hasKey, ['implicit', 'password', 'clientCredentials', 'authorizationCode']));
+const grantTypes = {
+  implicit: 'implicit',
+  password: 'resource owner password credentials',
+  clientCredentials: 'client credentials',
+  authorizationCode: 'authorization code',
+};
+
+/**
+ * Parse Oauth Flows Object
+ *
+ * @param namespace {Namespace}
+ * @param element {Element}
+ * @returns ParseResult
+ *
+ * @see https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#oauthFlowsObject
+ * @private
+ */
+function parseOauthFlowsObject(context, object) {
+  const { namespace } = context;
+
+  const parseFlow = (member) => {
+    const key = member.key.toValue();
+    const flow = parseOauthFlowObject(context, member.value);
+
+    if (flow.length === 0 || isAnnotation(flow.get(0))) {
```

Comment:
Can I use `pipeParseResult` here?

Reply:
Perhaps:

```js
pipeParseResult(namespace,
  parseOauthFlowObject(context),
  R.when(R.anyPass('implciit', 'authorizatioNCode' ..., createWarning()...)
  R.when(R.anyPass('password', 'clientCredentials', 'authorizationCode' ..., createWarning()...)
```

Very pseudocode, but you get the idea.